src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -241,6 +349,43 @@ else if (!period.isEmpty()) {
         }
     }
 
+    /**
+     * Check the summary of inline form.
+     *
+     * @param ast Javadoc root Node.
+     */
+    public void validateSummaryTag(DetailNode ast) {
+        final String inlineSummaryDoc = getInlineSummarySentence(ast);
+        if (inlineSummaryDoc.isEmpty()) {
+            log(ast.getLineNumber(), MSG_SUMMARY_JAVADOC_MISSING);
+        }
+        else if (!period.isEmpty()) {
+            if (!inlineSummaryDoc.contains(period)) {
+                log(ast.getLineNumber(), MSG_SUMMARY_FIRST_SENTENCE);
+            }
+
+            if (isFirstSentenceAllowed(ast)) {
```

Comment:
I do not understand why `inlineSummaryDoc` is not placed in this method and you do calculation of it one more time.
it either getInlineSummarySentence or getInlineFirstSentence do bad job, but there should single implementation on how to get summary sentence.

Reply:
Sir, here the implementation is similar to [visitJavadocToken](https://github.com/checkstyle/checkstyle/blob/ebef3bd59fdc2c97587a67bc59d57af6ecbb9072/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java#L224).

If we have a javadoc similar to 
```
    /**
     * {@summary first sentence is normally the summary.
     * Use of html tags:
     * <ul>
     * <li>Item one.</li>
     * <li>Item two.</li>
     * </ul>}
     */
    private void validInlineJavadoc() {                   // ok
    }
```
What `getInlineSummarySentence()` would return is `first sentence is normally the summary. Use of html tags: <ul> <li>Item one.</li> <li>Item two.</li> </ul>`, where as result of `getInlineFirstSentence()` is `first sentence is normally the summary.` . 

Here first we check if the there is any Javadoc or not by getting the entire the summary with help of `getInlineSummarySentence()`, if it is present we would like to know, if there is any `PERIOD` present in string returned by `getInlineSummarySentence()`, to confirm that it is at least one complete sentence.  Next we get the first sentence and check if there is any anomaly(Invalid tags and fragments) in it or not.

Reply:
what is point to return `Use of html tags: <ul> <li>Item one.</li> <li>Item two.</li> </ul>` test. All we care is first/summary sentence. In old style javadoc first and summary sentence is same.
In modern javadoc, with summary tag, it might be a bit different as it might mean all content from `{` until `}`. BUT this Check do not care about any other content of summary tag that is after first `.`.

Reply:
Sir, we do check the entire summary sentence. In method `isFirstSentenceAllowed(ast)` it checks that `containsCorrectInlineTags(ast)` is true or not, and the `containCorrectInlineTags(ast)` checks the entire summary sentence.

Eg;- 
```
    /**                                                 // violation for Forbidden fragment
     * {@summary first sentence is normally the summary.
     * Use of html tags:
     * {@throws error}
     * <ul>
     * <li>Item one.</li>
     * <li>Item two.</li>
     * </ul>}
     */
    private void validInlineJavadoc() // violation
    {
    }
```
This example has been added to `InputSummaryJavadocInlineForbidden.java` , also I had changed the summary for `containCorrectInlineTags(ast)` and renamed the `isFirstSentenceAllowed(ast)` to `isSummaryValid(ast)`  to make things a bit more clear. Sorry for the delay in response, I was checking if all is fine discovered [this](https://github.com/checkstyle/checkstyle/pull/9174#issuecomment-803623910).

Reply:
we do not need first Sentence in `summary` tag content. Whole tag content is summary so at the end of whole content there should be period, even it is miltiline of multi-sentence.

Please see example at  https://github.com/checkstyle/checkstyle/issues/7202#issuecomment-766890211

Reply:
Done, refracted check. Sample output
```
$ cat Test.java
cat Test1.java    
 public class Test1{
 
    /**
     * {@summary first sentence is normally the summary.
     * Use of html tags:
     * <ul>
     * <a href="NOEHRE">Item one.</a>
     * <a href="SOMEWEHRE">Item two</a>
     * </ul>}
     */
    public void validInlineJavadoc() // ok
    {
    }
 }

$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
<module name="SummaryJavadocCheck"/>
  </module>
</module>

$ java -jar checkstyle-8.42-SNAPSHOT-all.jar -c config.xml Test1.java                             
Starting audit...
[ERROR] /home/akmo/GitHub/Test/InlineJavadoc/Test1.java:3: First sentence of Javadoc is missing an ending period. [SummaryJavadoc]
Audit done.
Checkstyle ends with 1 errors.
```

Reply:
ok, looks like now I see root of confusion, it is validation message :).
`Test1.java:3: First sentence of Javadoc is missing an ending period.`
line 3 is:
`     * {@summary first sentence is normally the summary.`
it does have a period.
Lets create new message for Check for this case `Summary of Javadoc is missing an ending period.`

Reply:
Done.